include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -683,6 +783,123 @@ __pattern_transform_scan_multi_group(_ExecutionPolicy&& __exec, _InRng&& __in_rn
         unseq_backend::__global_scan_functor<_Inclusive, _BinaryOperation, _InitType>{__binary_op, __init});
 }
 
+template <typename _SizeType>
+struct __invoke_single_group_copy_if
+{
+    // Specialization for devices that have a max work-group size of at least 1024
+    static constexpr ::std::uint16_t __targeted_wg_size = 1024;
+
+    template <::std::uint16_t _Size, typename _ExecutionPolicy, typename _InRng, typename _OutRng, typename _Pred>
+    auto
+    operator()(_ExecutionPolicy&& __exec, ::std::size_t __n, _InRng&& __in_rng, _OutRng&& __out_rng, _Pred&& __pred)
+    {
+        constexpr ::std::uint16_t __wg_size = ::std::min(_Size, __targeted_wg_size);
+        constexpr ::std::uint16_t __num_elems_per_item = ::oneapi::dpl::__internal::__dpl_ceiling_div(_Size, __wg_size);
```

Comment:
Similar comment as above about tuning this parameter. Other algorithms have shown best performance with 32 elements per work item while this uses 16 at most, I think.

Reply:
I'll test out different parameters.

Reply:
I tested out a few different parameters and they were within 1-2% of each other with no clear winner. Based on those experiments, I think I will just keep it to 1024 for now.